gui/qt/console.py

```diff
@@ -35,6 +122,35 @@ def __init__(self, prompt='>> ', startup_message='', parent=None):
         self.updateNamespace({'run':self.run_script})
         self.set_json(False)
 
+        self.warningOverlay = None
+
+        wallet_storage = parent.wallet.storage
+        config_dontaskagain = wallet_storage.get(self.CONFIG_DONTASKAGAIN_KEY, False)
+        config_relaxwarn = get_config().cmdline_options["relaxwarn"]
+
+        # Don't show the warning if the user chose to have it not shown again or
+        # when the relaxwarn option is set.
+        if not config_dontaskagain and not config_relaxwarn:
+            self.warningOverlay = ConsoleWarningOverlay(self)
+            self.warningOverlay.resize(self.size())
+
+            fp = self.focusPolicy()
+            self.setFocusPolicy(QtCore.Qt.NoFocus)
+            self.setFocusProxy(self.warningOverlay)
+            def on_acknowledged(dontaskagain: bool):
+                wallet_storage.put(self.CONFIG_DONTASKAGAIN_KEY, dontaskagain)
+                self.warningOverlay.deleteLater()
```

Comment:
I really appreciate your paying attention to object lifecycle.  The Electrum programmers never did and it took me ages to track down memory leaks.

I have found in my own experiments that `deleteLater()` is dangerous and more trouble than it's worth.  The problem is there isn't a strict 1:1 correlation between C++ deleting an object and Python realizing it's gone.

You are better off relying on the Python GC here rather than calling `deleteLater`.  

What normally works:

```
  someObj.setParent(None)
  someObj = None  # at this point refct may drop to 0, and it's immediately deleted
```

Python GC's QObjects with no Parent.

It may keep the object around though if a lambda is still referring to it from a callback or from a timer.  This is why deleteLater is dangerous. You get a "C++ object deleted" runtimeerror... You're better off relying on the python GC.

If you want to track if it really was deleted -- you can register the object finalizer:

`util.finalization_print_error(obj)`

There's also a:

`electroncash_gui.qt.util.destroyed_print_error(obj)`  thing to register `destroyed` signals from Qt but those don't always fire to python for some reason.

Reply:
That is good to know, thanks. I'll adjust accordingly.